GNUmakefile

```diff
@@ -240,22 +240,10 @@ generate-structs: ## Update generated code
 	@echo "--> Running go generate..."
 	@go generate $(LOCAL_PACKAGES)
 
-## The ",M<path/to/proto>=<package name>" below is required to tell
-## protoc-gen-go what the import path for a given proto file should be. This is
-## necessary when a proto file foo/a.proto imports a proto file from another
-## directory, e.g. bar/b.proto
 .PHONY: proto
 proto:
 	@echo "--> Generating proto bindings..."
-	@for file in $$(git ls-files "*.proto" | grep -E -v -- "vendor\/.*.proto|demo\/.*.proto"); do \
```

Comment:
This can actually be problematic - it's an edge case, but `protoc` (at least in the past, they might have fixed this by now but unlikely) had different results if you compiled per-file as opposed to all the files in one directory. Generally, you'd want to do something along the lines of:

```bash
for dir in $(find . -name '*.proto' -print0 | xargs -0 -n1 dirname | sort | uniq); do
  protoc -I . -o /dev/null $(find "${dir}" -name '*.proto')
done
```

Reply:
Of course, we at Buf have thought about this, so you don't have to worry about this problem with `buf generate` :-)

Reply:
This is similar to what we ended up doing on another internal HashiCorp repo while beginning the process of switching away from `prototool`. Because we were using `prototool` with 4 plugins and multiple proto roots, the actual implementation was far more elaborate.

I actually started with the snippet you have above but it ended up boiling down to this:

```bash
    for protoDir in $protoDirs; do
      protoFiles=$(find "$protoDir" -name '*.proto' | sort | tr '\n' ' ')

      for plugin in $plugins; do
        cmd="$protocPath $includes $plugin $protoFiles"
        echo "$cmd"
        $cmd
      done
    done
```

Where the `$includes` and `$plugin` are all the wild `-I` flags and plugin-specific invocation+flags for each plugin we use.

Reply:
Sounds about right :-)